### DISCUSSION TITLE
How could I change tabs using the SHIFT key?

### DISCUSSION THREAD

**Discussion by rcardoso20** [vscode-neovim/vscode-neovim#2226] [General]
On neovim I usually change tabs with H and L, how could I replicate this with the vscode neovim extension?

I tried this on my init.lua, but with no success:
``` lua
vim.keymap.set('n', 'L', require('vscode').call("workbench.action.previousEditor"), {noremap = true, silent = true})
vim.keymap.set('n', 'H', require('vscode').call("workbench.action.nextEditor"), {noremap = true, silent = true})
```

When on VSCode, I run the command following command: verbose map H
It returns the following:
```
x  H           * <Cmd>call <SNR>9_moveCursor('top')<CR>
	Last set from ~/.vscode/extensions/asvetliakov.vscode-neovim-1.18.10/runtime/vscode/overrides/vscode-scrolling.vim line 26
n  H           * :bprevious<CR>
	Last set from Lua (run Nvim with -V1 for more details
```

I also tried to unmap these 2 keybinds on my init.lua, but also with no success.

Thanks for the help!

**Comment by ian-h-chamberlain** [comment]
The extension registers those keymaps in its `overrides` (as shown by the output you pasted), which happens after your `init.lua` is sourced, so I guess without a change to vscode-neovim you won't able to set a vim keymap.

However, you can probably configure VSCode to bypass vscode-neovim entirely, e.g. in `keybindings.json`:

```jsonc
    {
        "key": "shift+l",
        "command": "workbench.action.previousEditor",
        "when": "editorTextFocus && neovim.mode == 'normal'",
    },
    {
        "key": "shift+h",
        "command": "workbench.action.nextEditor",
        "when": "editorTextFocus && neovim.mode == 'normal'",
    }
```
From a quick test this seems to work for me at least.

**Reply by justinmk** [reply]
> The extension registers those keymaps in its `overrides` (as shown by the output you pasted), which happens after your `init.lua` is sourced

I wonder why we chose to do that *after* the user's config. That seems like a bug.

**Reply by ian-h-chamberlain** [reply]
Hmm, maybe I'm mistaken about the order of things getting sourced... I think I was thinking more of `runtime/lua/vscode/force-options.lua`, I guess the issue was actually just using a callback; this also works on my setup:
```lua
    vim.keymap.set("n", "L", function()
        require("vscode").call("workbench.action.previousEditor")
    end, { noremap = true, silent = true })
    vim.keymap.set("n", "H", function()
        require("vscode").call("workbench.action.nextEditor")
    end, { noremap = true, silent = true })
```

**Comment by xiyaowong** [comment]
You didn't pass the function to 'vim.keymap.set', but passed the function result.

**Comment by rcardoso20** [comment]
Thanks for your asnwers, but I still couldn't make it work.
Still has the same behavior.

When I press SHIFT+H or SHIFT+L the cursor moves to the top of the document and gets stuck there, I can't even move down with J, to get it to unstuck I need to press SHIFT+H or SHIFT+L again and it returns back to the position I was before and then it allows me to move the cursor freely.

**Comment by rcardoso20** [comment]
I found out what the problem was...holy moly.
So, from the beginning... I was going to use vscode instead of neovim and since I already had a full configuration for it I created this on my init.lua:

``` lua
if vim.g.vscode then
    require("user.vscode")
else
    require('user.core')
    require('user.plugins')
end
```
My user/vscode/init.lua has the following:

``` lua
local core = 'user.core'

require(core .. '.options')
require(core .. '.keymaps')
require(core .. '.autocmds')
```
The problem here is that I my variable "core" was going back to "user.core" instead of "vscode", so I was actually loading all the stuff from my "normal" neovim config...
Since I was using the same keybinds, I didn't notice that the changes I was doing wouldn't take effect, so I could be trying to this for ages and it would never work...

I apologize for the the mistake guys, although, I'm thankful for you helping me with the function because that problem would still exist :D
